server/controllers/commands/request.js

```diff
@@ -0,0 +1,103 @@
+const dataworld = require('../../api/dataworld')
+const slack = require('../../api/slack')
+const { InvalidCaseError } = require('../../helpers/errors')
+const { AUTHORIZATION_ACTIONS } = require('../../helpers/requests')
+const { getBotAccessTokenForChannel } = require('../../helpers/tokens')
+
+const openNotificationModal = async (channelId, triggerId, message) => {
+  const token = await getBotAccessTokenForChannel(channelId)
+  const modalView = {
+    type: 'modal',
+    title: {
+      type: 'plain_text',
+      text: 'Something went wrong 😞'
+    },
+    blocks: [
+      {
+        type: 'section',
+        text: {
+          type: 'mrkdwn',
+          text: message
+        }
+      }
+    ]
+  }
+  slack.openView(token, triggerId, modalView)
+}
+
+const updateMessageBlocksWithAction = (messageBlocks, blockId, userId, action) => {
+  const buttonsIndex = messageBlocks.findIndex(block => block.block_id === blockId)
+  messageBlocks[buttonsIndex] = {
+    type: 'section',
+    text: {
+      type: 'mrkdwn',
+      text: `*Request ${action} by <@${userId}>*`
+    }
+  }
+  return messageBlocks
+}
+
+const handleDatasetRequestAction = async ({
+  channelid,
+  userid,
+  triggerid,
+  responseUrl,
+  message,
+  blockid,
+  actionid,
+  requestid,
+  agentid,
+  datasetid,
+  dwAccessToken
+}) => {
+  let action
+  try {
+    switch (actionid) {
+      case AUTHORIZATION_ACTIONS.ACCEPT:
+        await dataworld.acceptDatasetRequest(dwAccessToken, requestid, agentid, datasetid)
+        action = 'approved'
+        break
+      case AUTHORIZATION_ACTIONS.REJECT:
+        await dataworld.rejectDatasetRequest(dwAccessToken, requestid, agentid, datasetid)
+        action = 'rejected'
+        break
+      case AUTHORIZATION_ACTIONS.CANCEL:
+        await dataworld.cancelDatasetRequest(dwAccessToken, requestid, agentid, datasetid)
+        action = 'cancelled'
+        break
+      default:
+        throw new InvalidCaseError(actionid)
+    }
+  } catch (error) {
+    if (error.response && error.response.status === 403) {
+      await openNotificationModal(
+        channelid,
+        triggerid,
+        'You are not authorized to manage this request.'
+      )
+    } else {
+      await openNotificationModal(
+        channelid,
+        triggerid,
+        'Could not successfully manage this request. Try viewing the request on <https://data.world|data.world>.'
```

Comment:
nit: this won't be an issue right now, but this hard-coded url will need to change eventually.

Reply:
A lot of the pre-existing features rely on hard-coded `data.world` URLs (e.g. link previews and dataset/project subscriptions). It might be good to handle these all at once, once we know more about how the sites functionality will be implemented.

Reply:
Yep agreed.